Releasing tasks even if something failed while parsing

stratg5 opened the pull request:
It appears as though all other calls to `ToIDispatch()` have a `Release()` call immediately after. I noticed that for tasks, this is not the case, and if there is an error while parsing a task, we will not release the task. Opening a PR to release tasks, even if something goes wrong with parsing.

If this is not correct please let me know! But I believe that even if there is an error, we still need to release, as we have already obtained the memory.

Commit:
Releasing tasks even if something failed while parsing

Commit:
Parse the XML text

Commit:
Pull xmltext from definition instead of actions

giert commented:
It seems to be an error, yes. Good catch! I suspect this is not the last potential memory leak to be found, but we'll get there eventually.

I've changed this method to return err rather than relying on panics in the parsing, so just do the same for xml strings and I'll happily merge it in. It's nice that there's more people looking at this important, but sadly error-prone library.

stratg5 commented:
> It seems to be an error, yes. Good catch! I suspect this is not the last potential memory leak to be found, but we'll get there eventually.
> 
> I've changed this method to return err rather than relying on panics in the parsing, so just do the same for xml strings and I'll happily merge it in. It's nice that there's more people looking at this important, but sadly error-prone library.

Awesome! Thanks for making that change, I think returning an error instead of panic is a good thing too. I've updated the `XmlText` to follow that pattern. I'm happy we are not alone also 😆

giert commented:
Perfect! My goal is to gradually transition the library from relying on panics to return err in cases where it would make more sense to do so. I will do this gradually however, as I need to be able to determine if there's any unintended side effects of these changes.
